### DISCUSSION TITLE
MIDI jitter

### DISCUSSION THREAD

**Discussion by davekc** [lathoub/Arduino-AppleMIDI-Library#153] [Q&A]
Hello everyone!

I don’t have the skills to decipher the [RFC4695](https://www.rfc-editor.org/rfc/rfc4695) or even the [RFC6295](https://www.rfc-editor.org/rfc/rfc6295) specifications and understand if the AppleMIDI (aka rtpMIDI) is subject to MIDI jitter or not.

Seeing the latency gauge in the MacOS MIDI Network Setup or the piece of code below, makes me feel like yes (considering that latency can vary). But maybe I’m wrong. Can someone enlighten me?

```
  AppleMIDI.setHandleReceivedRtp([](const APPLEMIDI_NAMESPACE::ssrc_t & ssrc, const APPLEMIDI_NAMESPACE::Rtp_t & rtp, const int32_t& latency) {
    AM_DBG(F("setHandleReceivedRtp"), ssrc, rtp.sequenceNr , "with", latency, "ms latency");
  });
```

Kind regards,
Dave

**Comment by lathoub** [comment]
Using Wifi or Ethernet?
(wifi causes a lot of jitter, see https://github.com/lathoub/Arduino-AppleMIDI-Library/wiki/Keeping-Latency-under-control)

**Comment by davekc** [comment]
Ethernet, I should have straightened it out...

**Comment by lathoub** [comment]
MIDI messages uses UDP and is not reliable, that said, i never had messages being dropped because of UDP.
Most of time, too many MIDI message are being send at once, causing overflow - hence you need to check what Wireshark gives.
Also, I have had message drops reported because of DAW issues.
In any case, a Wireshark dump is needed and will point to the issue

**Reply by davekc** [reply]
Maybe this reply concerns with my other question? I will check all that for sure. Thanks!

Here its more a generic question... does the RTP approach causes any MIDI jitter like what is found with USB MIDI devices, for instance.

**Comment by lathoub** [comment]
> does the RTP approach causes any MIDI jitter like what is found with USB MIDI devices, for instance.

I never had jitter for USB MIDI, unless the USB MIDI device was not good (**never** had an issue with Roland USB MIDI devices).
RTP is normally very reliable, the issue is probably because too many MIDI commands are send at once.
